src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs

```diff
@@ -88,20 +80,21 @@ public Task MarkAsCompleted()
             return TaskEx.CompletedTask;
         }
 
-        public object Read()
+        public async Task<object> Read()
         {
-            return serializer.Deserialize(jsonReader, manifest.SagaEntityType);
+            var json = await streamReader.ReadToEndAsync().ConfigureAwait(false);
+            return SimpleJson.DeserializeObject(json, manifest.SagaEntityType, SerializationStrategy);
         }
 
 
         SagaManifest manifest;
         FileStream fileStream;
         bool isCompleted;
-        JsonTextWriter jsonWriter;
-        JsonTextReader jsonReader;
+        StreamWriter streamWriter;
+        StreamReader streamReader;
 
         const int DefaultBufferSize = 4096;
-        static Newtonsoft.Json.JsonSerializer serializer = new Newtonsoft.Json.JsonSerializer();
         static Task<SagaStorageFile> noSagaFoundResult = Task.FromResult<SagaStorageFile>(null);
+        static readonly IJsonSerializerStrategy SerializationStrategy = new EnumAwareStrategy();
```

Comment:
Since we're using SimpleJson in a couple of unrelated places, it would be weird to use the `CurrentJsonSerializerStrategy` property, since setting it in one place would mean it would be used globally.

I'm OK with the current approach, though have a single static in the `EnumAwareStrategy` class sounds OK too.